assets/js/extensions/google-analytics/index.ts

```diff
@@ -13,89 +17,217 @@ import {
 	getProductFieldObject,
 	getProductImpressionObject,
 	trackEvent,
+	trackCheckoutStep,
+	trackCheckoutOption,
 } from './utils';
 
-const trackListProducts = ( {
-	products,
-	listName = __( 'Product List', 'woo-gutenberg-products-block' ),
-}: {
-	products: Array< ProductResponseItem >;
-	listName: string;
-} ): void => {
-	trackEvent( 'view_item_list', {
-		event_category: 'engagement',
-		event_label: __( 'Viewing products', 'woo-gutenberg-products-block' ),
-		items: products.map( ( product, index ) => ( {
-			...getProductImpressionObject( product, listName ),
-			list_position: index + 1,
-		} ) ),
-	} );
-};
-
-const trackAddToCart = ( {
-	product,
-	quantity = 1,
-}: {
-	product: ProductResponseItem;
-	quantity: number;
-} ): void => {
-	trackEvent( 'add_to_cart', {
-		event_category: 'ecommerce',
-		event_label: __( 'Add to Cart', 'woo-gutenberg-products-block' ),
-		items: [ getProductFieldObject( product, quantity ) ],
-	} );
-};
-
-const trackRemoveCartItem = ( {
-	product,
-	quantity = 1,
-}: {
-	product: CartResponseItem;
-	quantity: number;
-} ): void => {
-	trackEvent( 'remove_from_cart', {
-		event_category: 'ecommerce',
-		event_label: __( 'Remove Cart Item', 'woo-gutenberg-products-block' ),
-		items: [ getProductFieldObject( product, quantity ) ],
-	} );
-};
+/**
+ * Track customer progress through steps of the checkout. Triggers the event when the step changes.
+ *
+ * 1 - Contact information
+ * 2 - Shipping address
+ * 3 - Billing address
+ * 4 - Shipping options
+ * 5 - Payment options
```

Comment:
On line 76 you send a 6 for step and I don't see step 5 being send anywhere
```
trackCheckoutOption( {
			step: 6,
			option: __( 'Payment Method', 'woo-gutenberg-products-block' ),
			value: paymentMethodSlug,
		} )();
```

Reply:
Fixed in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3967/commits/0bab0de1e09cadf3f04b123b36d625b54b8c09ea